org.eclipse.xtext.xbase.junit/src/org/eclipse/xtext/xbase/junit/typesystem/Oven.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2013, 2020 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.xtext.xbase.junit.typesystem;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.function.Consumer;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.xtext.common.types.JvmFormalParameter;
+import org.eclipse.xtext.common.types.JvmIdentifiableElement;
+import org.eclipse.xtext.junit4.util.ParseHelper;
+import org.eclipse.xtext.xbase.XAbstractFeatureCall;
+import org.eclipse.xtext.xbase.XClosure;
+import org.eclipse.xtext.xbase.XExpression;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+import org.eclipse.xtext.xbase.lib.IteratorExtensions;
+import org.eclipse.xtext.xbase.lib.util.ReflectExtensions;
+import org.eclipse.xtext.xbase.typesystem.IBatchTypeResolver;
+import org.eclipse.xtext.xbase.typesystem.IResolvedTypes;
+import org.eclipse.xtext.xbase.typesystem.internal.CompoundReentrantTypeResolver;
+import org.eclipse.xtext.xbase.typesystem.internal.RootResolvedTypes;
+import org.eclipse.xtext.xbase.typesystem.internal.TypeData;
+import org.eclipse.xtext.xbase.typesystem.references.LightweightTypeReference;
+import org.junit.Assert;
+import org.junit.ComparisonFailure;
+
+import com.google.common.annotations.Beta;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * Utility to check a given expression for exceptions and integrity after it was processed by the {@link IBatchTypeResolver}
+ * 
+ * @author Sebastian Zarnekow - Initial contribution and API
+ * @deprecated Use org.eclipse.xtext.xbase.testing.typesystem.Oven instead
+ */
+@Deprecated
+@Beta
+@Singleton
+public class Oven extends Assert {
+	@Inject
+	private IBatchTypeResolver typeResolver;
+
+	@Inject
+	private ReflectExtensions reflectExtensions;
+
+	@Inject
+	private ParseHelper<EObject> parseHelper;
+
+	public void fireproof(CharSequence input) throws Exception {
+		fireproof(input.toString());
+	}
+
+	public void fireproof(String input) throws Exception {
+		try {
+			EObject file = parseHelper.parse(input);
+			IResolvedTypes resolvedTypes = typeResolver.resolveTypes(file);
+			Assert.assertNotNull(resolvedTypes);
+			if (file != null) {
+				for (EObject content : IteratorExtensions.toIterable(file.eAllContents())) {
```

Comment:
Ideally we use something other than IteratorExtensions

Reply:
dont know an alternative